crates/containerd-shim-wasm/src/sys/unix/container/instance.rs

```diff
@@ -109,30 +108,20 @@ impl<E: Engine> SandboxInstance for Instance<E> {
     fn delete(&self) -> Result<(), SandboxError> {
         log::info!("deleting instance: {}", self.id);
         match instance_exists(&self.rootdir, &self.id) {
-            Ok(exists) => {
-                if !exists {
-                    return Ok(());
-                }
-            }
+            Ok(true) => {}
+            Ok(false) => return Ok(()),
             Err(err) => {
                 log::error!("could not find the container, skipping cleanup: {}", err);
                 return Ok(());
             }
         }
         let container_root = get_instance_root(&self.rootdir, &self.id)?;
-        let container = Container::load(container_root)
-            .with_context(|| format!("could not load state for container {}", self.id));
-        match container {
-            Ok(mut container) => container.delete(true).map_err(|err| {
-                SandboxError::Any(anyhow::anyhow!(
-                    "failed to delete container {}: {}",
-                    &self.id,
-                    err
-                ))
-            })?,
+        match Container::load(container_root) {
+            Ok(mut container) => {
+                let _ = container.delete(true);
```

Comment:
are we losing the fact that it might fail here?

Reply:
Indeed, I missed that.